### ISSUE TITLE
Unable to generate links / actions in CLI mode

### ISSUE THREAD

**Issue by SC-cmarquardt** [neos/flow-development-collection#2084]
### Description
Currently I am unable to create a link or an URI with `f:link.action` (or corresponding URI method) on a Windows server in a CommandController. I set `Neos.Flow.http.baseUri` but for some reason the ViewHelper creates a local file-based path, which means s.th. like `E:\my\path\to\flow\index.php`.
Is this a know issue? Also setting the `absolute` parameter to false for the ViewHelper does not change anything.

### Used Versions

Flow: 6.1.4

**Comment by sorenmalling** [comment]
Hi @SC-cmarquardt 

The issue and topic have been raised in this former ticket

https://github.com/neos/flow-development-collection/issues/1848

Will you try and see if the provided examples and referred links helps you? If not, please open a thread on discuss.neos.io so we can get more people involved in the debugging. If the issue remain, this issue ticket is absolutely valid :-)

**Comment by SC-cmarquardt** [comment]
Hi @sorenmalling,

thanks for your reply.
Christian (kitsunet) writes here https://github.com/neos/flow-development-collection/issues/1848#issuecomment-554409888 that in Flow 6 things behave differently and it should be easier to create a "fake" request.
Do you have an example for this fake request?

**Comment by sorenmalling** [comment]
@SC-cmarquardt 

I don't know of such thing - I personally create links in a more "static way" (typing the url and using a setting for root domain) to avoid the CLI issue with trying to create urls.

I suggest that you create a ticket on discuss.neos.io with this question - more will probably benefit from it and can help on solving this

**Comment by sorenmalling** [comment]
Did you try the example Christian provides https://github.com/neos/flow-development-collection/issues/1848#issuecomment-554410993

**Comment by albe** [comment]
The most "correct" way as of Flow 6.0 to get a fake `ActionRequest` which you can pipe into the StandaloneView would be like this:
```php

/**
 * @Flow\Inject
 * @var ActionRequestFactory
 */
protected $actionRequestFactory;

/**
 * @Flow\Inject
 * @var \Neos\Http\Factories\ServerRequestFactory
 */
protected $serverRequestFactory;

...

$fakeHttpRequest = $this->serverRequestFactory->createServerRequest('GET', 'https://acme.com/your/fake/uri', $optionalHTTParguments);
$fakeActionRequest = $this->actionRequestFactory->createActionRequest($fakeHttpRequest, []);
$view = new StandaloneView($fakeActionRequest);
```
The `ActionRequestFactory` does a tiny bit more than just `ActionRequest::fromHttpRequest($fakeHttpRequest)`, but for CLI it would likely not make a realy difference.
You could get the `your/fake/uri` from the configured `Neos.Flow.http.baseUri` setting (inject manually or use `BaseUriProvider::getConfiguredBaseUri()` - see also #1755) instead of hardcoding one.

What I found though, is that as of Flow 6.0.x the UriBuilder (which creates the URIs in the Views) uses the configured baseUri by default and only falls back to the Http request URI as base: https://github.com/neos/flow-development-collection/pull/1755/files#diff-fcf7039aeecafa562dd6ad82a86bb3b4R361
So I'm not sure why that doesn't work here. Maybe we need to have a functional test or minimal setup showing the wrong behaviour and play around.

Anyway, we definitely need to find a better way to handle the relatively common concern of rendering Views from the CLI. Maybe we just create a default `StandaloneView`, or at least a default fake `HttpRequest` (though I don't like that so much, because what is a sane "HttpRequest" that reflects a CLI request?)

**Comment by bwaidelich** [comment]
The regression came with #1755 since `BaseUriProvider::getConfiguredBaseUriOrFallbackToCurrentRequest()` just fails with an exception if no `baseUri` is configured
I just talked to @kitsunet and we found a (temporary) solution for this problem:
I would suggest to extend the method by an optional fallback request like: `BaseUriProvider::getConfiguredBaseUriOrFallbackToCurrentRequest(ServerRequestInterface $fallbackRequest = null)`
(similar to https://github.com/neos/neos-development-collection/blob/master/Neos.Fusion/Classes/Eel/BaseUriHelper.php#L31) and specify that in the `UriBuilder` (and in other relevant parts, e.g. the Neos `LinkingService`).

I'll provide a PR for branch 6.0 (and Neos 5.0).

For 7.0+ I would like to get rid of the `baseUri` concept all together (see #2157)
